block-storage/src/main/scala/coop/rchain/blockstorage/LMDBBlockStore.scala

```diff
@@ -0,0 +1,128 @@
+package coop.rchain.blockstorage
+
+import java.nio.ByteBuffer
+import java.nio.file.Path
+
+import scala.collection.JavaConverters._
+import scala.language.higherKinds
+
+import cats._
+import cats.effect.{ExitCase, Sync}
+import cats.implicits._
+import com.google.protobuf.ByteString
+import coop.rchain.blockstorage.BlockStore.BlockHash
+import coop.rchain.casper.protocol.BlockMessage
+import coop.rchain.metrics.Metrics
+import org.lmdbjava._
+import org.lmdbjava.DbiFlags.MDB_CREATE
+
+class LMDBBlockStore[F[_]] private (val env: Env[ByteBuffer], path: Path, blocks: Dbi[ByteBuffer])(
+    implicit
+    syncF: Sync[F],
+    metricsF: Metrics[F])
+    extends BlockStore[F] {
+
+  import LMDBBlockStore.MetricNamePrefix
+
+  implicit class RichBlockHash(byteVector: BlockHash) {
+
+    def toDirectByteBuffer: ByteBuffer = {
+      val buffer: ByteBuffer = ByteBuffer.allocateDirect(byteVector.size)
+      byteVector.copyTo(buffer)
+      // TODO: get rid of this:
+      buffer.flip()
+      buffer
+    }
+  }
+
+  def put(f: => (BlockHash, BlockMessage)): F[Unit] =
+    for {
+      _ <- metricsF.incrementCounter(MetricNamePrefix + "put")
+      ret <- syncF.bracket(syncF.pure(env.txnWrite())) { txn =>
```

Comment:
I think that acquisition should also be done with `syncF.delay { … } `

Reply:
What about releasing?